### DISCUSSION TITLE
Laravel Validation: Distinct Multiple Fields

### DISCUSSION THREAD

**Discussion by susutw** [laravel/framework#47642] [Ideas]
I am trying to valid the array, like the example:
```

    'product' => [
        [
            'type' => 'ticket',
            'name' => 'train',
            'price' => 100,
        ],
        [
            'type' => 'ticket',
            'name' => 'train',
            'price' => 200,
        ],
        [
            'type' => 'toy',
            'name' => 'train',
            'price' => 100,
        ]
    ]
```
I hope it can have a rule "Distinct Multiple Fields", and I can use the rule like `'product.*' => 'distinct_multiple_fields:type,name'`.
Giving the specified fields, and you can valid the value in fields.
So if "type + name" value is duplicate (like the example "type:ticket, name:train"), the validation result will be fail.
But the example "type:ticket, name:train" and "type:toy, name:train" will pass the validation.

If someone who think it works, and I will try to implement the code, thanks. 🙏
And if you have any naming suggestion, please tell me!

**Comment by jjjrmy** [comment]
I like this idea!

**Comment by henzeb** [comment]
There is already 'distinct'. Which could be adapted. I hate long underscored names.

**Reply by jjjrmy** [reply]
or something like `Rule::distinct(['type', 'name'])`

**Reply by susutw** [reply]
Thanks for your reactions!
I will design using distinct :)

**Comment by jjjrmy** [comment]
```php
$this->collect('products')
   ->map(fn (array $product) => Arr::only($product, ['type', 'name']))  
   ->duplicates()->isEmpty()
```

**Comment by in-kajal-baldha** [comment]
array:4 [
    0 => array:2 [
      "master_report_type" => "CALF"
      "body_part" => "RIGHT"
    ]
    1 => array:2 [
      "master_report_type" => "CALF"
      "body_part" => "RIGHT"
    ]
    2 => array:2 [
      "master_report_type" => "CALF"
      "body_part" => "LEFT"
    ]
    3 => array:2 [
      "master_report_type" => "ELBOW"
      "body_part" => "RIGHT"
    ]
  ]

**We make custom validation rule " distinct_multiple_fields "  for check duplicate value in array.**


`<?php

namespace App\Rules;

use Illuminate\Contracts\Validation\Rule;

class DistinctMultipleFields implements Rule
{
      public  $params;
    /**
     * Create a new rule instance.
     *
     * @return void
     */

    public function __construct($params)
    {
        $this->params = $params;
    }

    /**
     * Determine if the validation rule passes.
     *
     * @param  string  $attribute
     * @param  mixed  $value
     * @return bool
     */
    public function passes($attribute, $value)
    {

       $collection = collect($this->params);
       $duplicate_data = $collection->duplicates();
       if(count($duplicate_data) > 0) {
           return false;
       }
           return true;
    }

    /**
     * Get the validation error message.
     *
     * @return string
     */
    public function message()
    {
        return 'Duplicate value is not allow in body part and side.';
    }
}
`

**How to use:**

` $request->validate([
            'report.*.master_report_type'=>['required',new  DistinctMultipleFields($this->report)],
]);`
